### ISSUE TITLE
[rule suggestion] prefer property access expression

### ISSUE THREAD

**Issue by aervin** [palantir/tslint#3566]
Depending on the code base and project objectives, the element access syntax is rarely a necessary approach. Also, (and someone correct me if I'm wrong) Intellisense helps programmers when using the property access syntax, but offers no hints with the element access syntax. Would be nice to have a rule for enforcing which style of access the project allows. 

Would lint against this:
```javascript
const foo = { bar: 'hello!' };
foo['bar'] = 'world!';
```

Correcting to this:
```javascript
const foo = { bar: 'hello!' };
foo.bar = 'world!'; 
```

**Comment by ajafff** [comment]
Have you tried `no-string-literal`? That seems to be exactly what you are asking for, although the name suggests it does something else.

**Comment by aervin** [comment]
@ajafff that is exactly what I was after. Thanks!
